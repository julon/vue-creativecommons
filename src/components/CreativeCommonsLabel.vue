<template lang="pug">
.creative-commons-label
  span.creative-commons-label--prefix This work is licensed under a 
  a.creative-commons-label--link(rel="license" :href="license.link") {{ license.name }}
  span.creative-commons-label--suffix .
</template>

<script>
import CreativeCommonsMixin from "./CreativeCommonsMixin";

export default {
  name: "CreativeCommonsLabel",
  mixins: [CreativeCommonsMixin]
};
</script>
