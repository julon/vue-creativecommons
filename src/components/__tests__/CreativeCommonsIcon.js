import Vue from "vue";
import CreativeCommonsLabel from "../CreativeCommonsIcon";

// helper function that mounts and returns the rendered text
function getRendered(Component, propsData) {
  const Constructor = Vue.extend(Component);
  const vm = new Constructor({ propsData: propsData }).$mount();
  return vm;
}

describe("CreativeCommonsIcon.vue", () => {
  const vm = getRendered(CreativeCommonsLabel, {
    allowAdaptations: "yes",
    allowCommercialUse: "yes",
    iconSize: "normal"
  });
  it("should render correct contents", () => {
    expect(vm.$el.querySelector(".creative-commons-icon--link").href).toEqual(
      "http://creativecommons.org/licenses/by/4.0/"
    );
    expect(vm.$el.querySelector(".creative-commons-icon--img").alt).toEqual(
      "Creative Commons Attribution 4.0 International License"
    );
  });
  it("should match the snapshot", () => {
    expect(vm.$el).toMatchSnapshot();
  });
});
