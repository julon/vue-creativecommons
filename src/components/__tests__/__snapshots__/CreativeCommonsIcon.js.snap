// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreativeCommonsIcon.vue should match the snapshot 1`] = `
<div
  class="creative-commons-icon"
>
  <a
    class="creative-commons-icon--link"
    href="http://creativecommons.org/licenses/by/4.0/"
    rel="license"
    target="_blank"
  >
    <img
      alt="Creative Commons Attribution 4.0 International License"
      class="creative-commons-icon--img"
      src="https://i.creativecommons.org/l/by/4.0/88x31.png"
    />
  </a>
</div>
`;
